include/RAJA/policy/sequential/reduce.hpp

```diff
@@ -60,461 +60,111 @@
-  {
-    Reduce()(m_val, m_idx, rhs, idx);
-    return *this;
-  }
-
-private:
-  //! pointer to the parent ReduceMinLoc object
-  const ReduceMinLoc *m_parent;
-  mutable T m_val;
-  mutable Index_type m_idx;
+  using Base = detail::BaseReduceMin<T, detail::ReduceSeq>;
+  using Base::Base;
 };
 
-/*!
- **************************************************************************
- *
- * \brief  Max reducer class template for use in sequential execution.
- *
- **************************************************************************
- */
 template <typename T>
 class ReduceMax<seq_reduce, T>
+    : public detail::BaseReduceMax<T, detail::ReduceSeq>
 {
-using Reduce = RAJA::reduce::max<T>;
-
 public:
-  //! prohibit compiler-generated default ctor
-  ReduceMax() = delete;
-
-  //! prohibit compiler-generated copy assignment
-  ReduceMax &operator=(const ReduceMax &) = delete;
-
-  //! compiler-generated move constructor
-  ReduceMax(ReduceMax &&) = default;
-
-  //! compiler-generated move assignment
-  ReduceMax &operator=(ReduceMax &&) = default;
-
-  //! constructor requires a default value for the reducer
-  RAJA_HOST_DEVICE explicit ReduceMax(T init_val)
-      : m_parent(nullptr), m_val(init_val)
-  {
-  }
-
-  //! create a copy of the reducer
-  /*!
-   * keep parent the same if non-null or set to current
-   */
-  RAJA_HOST_DEVICE ReduceMax(const ReduceMax &other)
-      : m_parent(other.m_parent ? other.m_parent : &other), m_val(other.m_val)
-  {
-  }
-
-  //! Destructor folds value into parent object.
-  RAJA_HOST_DEVICE ~ReduceMax()
-  {
-    if (m_parent) {
-      Reduce()(m_parent->m_val, m_val);
-    }
-  }
-
-  //! return the reduced min value.
-  /*!
-   *  \return the calculated reduced value
-   */
-  RAJA_HOST_DEVICE operator T() { return m_val; }
-
-  //! return the reduced min value.
-  /*!
-   *  \return the calculated reduced value
-   */
-  RAJA_HOST_DEVICE T get() { return operator T(); }
-
-  //! reducer function; updates the current instance's state
-  /*!
-   * Assumes each thread has its own copy of the object.
-   */
-  RAJA_HOST_DEVICE const ReduceMax &max(T rhs) const
-  {
-    Reduce()(m_val, rhs);
-    return *this;
-  }
-
-  //! reducer function; updates the current instance's state
-  /*!
-   * Assumes each thread has its own copy of the object.
-   */
-  RAJA_HOST_DEVICE ReduceMax &max(T rhs)
-  {
-    Reduce()(m_val, rhs);
-    return *this;
-  }
-
-private:
-  //! pointer to the parent ReduceMax object
-  const ReduceMax *m_parent;
-  mutable T m_val;
+  using Base = detail::BaseReduceMax<T, detail::ReduceSeq>;
+  using Base::Base;
 };
 
-/*!
- **************************************************************************
- *
- * \brief  Sum reducer class template for use in sequential execution.
- *
- **************************************************************************
- */
 template <typename T>
 class ReduceSum<seq_reduce, T>
+    : public detail::BaseReduceSum<T, detail::ReduceSeq>
```

Comment:
Can we actually replace detail::ReduceSeq with detail::Seq?  I just noticed it here, but if so, it would apply in other cases too.

Reply:
That seems reasonable, but I think we'd be best off making it reduce::detail::Seq if you'd like to do that.  Having something called Seq in the RAJA::detail namespace seems like asking for a future conflict.

Reply:
I was hoping there was a way to unify the traversal and reduction policies so there are fewer names for users to remember.  I suggest leaving it for now. Reordering the words and adding the scope operator between them doesn't help much.